newIDE/app/src/UI/SemiControlledAutoComplete.js

```diff
@@ -142,20 +142,17 @@ const filterFunction = (
 
 const handleChange = (
   event: SyntheticKeyboardEvent<HTMLInputElement>,
-  option: Option | string,
+  option: Option,
   props: Props
 ): void => {
   if (option.type !== 'separator') {
-    if (typeof option === 'string') props.onChange(option);
```

Comment:
This was really unnecessary. I had it in place while replacing downshift. I had a keybinding to 'Enter' which would let autocomplete accept any value. Now, we are using the blur event to accept arbitrary values, and even after adding onRequestClose, I'd like to keep using the blur event, for 2 reasons

1. It remains as close as possible to the real implementation of autocomplete :) Even material-ui's default behaviour is using the blur event, you can check it in the freeSolo demo.

2. It would be a huge headache in the future for developers while understanding, debugging and maintaining the unnecessary keybindings and events and at the same time, it adds little to no value in usability for users :)

Reply:
Ok thanks for doing this simplification :) Keeping the code as simple as possible is an important thing :)